whylabs_toolkit/monitor/diagnoser/models/diagnosis_report.py

```diff
@@ -0,0 +1,265 @@
+import pandas as pd
+from typing import Dict, List, Optional, Tuple
+from pydantic import BaseModel
+from whylabs_toolkit.monitor.models import (
+    Analyzer,
+    Monitor,
+    Segment,
+    TargetLevel,
+    FixedThresholdsConfig,
+    ConjunctionConfig,
+    DisjunctionConfig,
+    GlobalAction,
+)
+
+from whylabs_toolkit.monitor.diagnoser.helpers.describe import (
+    describe_truncated_table,
+    filter_by_index,
+    describe_truncated_list,
+)
+from whylabs_toolkit.monitor.diagnoser.helpers.utils import segment_as_readable_text
+
+
+class SegmentReport(BaseModel):
+    batchCount: int
+    segment: Segment
+    totalAnomalies: int
+    totalFailures: int
+    totalColumns: int
+
+
+class NamedCount(BaseModel):
+    name: str
+    count: int
+
+    def to_tuple(self) -> Tuple[str, int]:
+        return self.name, self.count
+
+
+class ConditionRecord(BaseModel):
+    columns: Optional[List[str]]  # not present for some conditions like stale analysis
+    info: Optional[Dict]
+    summary: str
+    name: str
+
+
+class QualityIssueRecord(BaseModel):
+    name: str
+    description: str
+    detectors: List[str]
+
+
+class ProfileSummary(BaseModel):
+    minRowName: str
+    minRowCount: int
+    maxRowName: str
+    maxRowCount: int
+
+    def describe(self) -> str:
```

Comment:
would it be useful to direct extension if any summary/diagnoser class should implement an abstract `describe`?

Reply:
rather hold off on this for now. Not 100% sure describe should be on each item anyway... the result if you compose descriptions hierarchically isnt really what we'd want.